Glimpse.ApplicationInsights.Sample/Controllers/HomeController.cs

```diff
@@ -28,6 +29,47 @@ public class HomeController : Controller
             var albumCount = GetTotalAlbumns(); 
             Trace.Write(string.Format("Total number of Albums = {0} and Albums with 'The' = {1}", albumCount.Item1, albumCount.Item2));
 
+            var telemetryClient = new Microsoft.ApplicationInsights.TelemetryClient();
+
+            //Sample Trace telemetry
+            TraceTelemetry traceSample = new TraceTelemetry();
+            traceSample.Message = "Slow response - database";
+            traceSample.SeverityLevel = SeverityLevel.Warning;
+            telemetryClient.TrackTrace(traceSample);
+
+            //Sample event telemetry
+            var properties = new Dictionary<string, string> { { "Property 1",string.Format("Album Count {0}" ,albumCount.Item1) } };
+            var measurements = new Dictionary<string, double> { { "Sample Meassurement", albumCount.Item1 } };
+            telemetryClient.TrackEvent("Top Selling Albums", properties, measurements);
+
+            //Sample exception telemetry
+            try
+            {
+                albumCount = null;
+                int count=albumCount.Item1;
+            }
+            catch (Exception ex)
+            {
+                telemetryClient.TrackException(ex, properties, measurements);
+            }
+
+            //Obtains the ip address from the request
+            var request = new RequestTelemetry();
+            request.Url = HttpContext.Request.Url;
+            string ip = HttpContext.Request.ServerVariables["HTTP_X_FORWARDED_FOR"];
+            if (string.IsNullOrEmpty(ip))
+            {
+                ip = HttpContext.Request.UserHostAddress;
+            }
+            else
+            {
+                ip = ip.Split(',').Last().Trim();
+            }
+            request.Success = false;
+            request.Context.Location.Ip = ip;
```

Comment:
You don't need to initializer ```Location.Ip``` - it will be initialized automatically by ```ClientIpHeaderTelemetryInitializer``` that you configured in ```ApplicationInsights.config``` file.